src/parser.rs

```diff
@@ -345,13 +345,20 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
               items.push(Item::Assignment(self.parse_assignment(false)?));
             } else {
               let doc = pop_doc_comment(&mut items, eol_since_last_comment);
-              items.push(Item::Recipe(self.parse_recipe(doc, false)?));
+              items.push(Item::Recipe(self.parse_recipe(doc, false, false)?));
             }
           }
         }
       } else if self.accepted(At)? {
         let doc = pop_doc_comment(&mut items, eol_since_last_comment);
-        items.push(Item::Recipe(self.parse_recipe(doc, true)?));
+        items.push(Item::Recipe(self.parse_recipe(doc, true, false)?));
+      } else if self.accepted(BracketL)? {
+        self.parse_attribute_name()?;
```

Comment:
This works, but it's a bit unclear what's going on. I'd do something like:

```rust
let Attribute::NoExitMessage = self.parse_atribute_name()?;
```

This makes it clear what's going on, and it will turn into a compile error when we add another variant.